src/auth/googleauth.ts

```diff
@@ -81,6 +81,12 @@ export interface GoogleAuthOptions {
    */
   clientOptions?: JWTOptions | OAuth2ClientOptions | UserRefreshClientOptions;
 
+  /**
+   * Scopes populated by the client library by default. We differentiate between
+   * these and user defined scopes when deciding whether to use a self-signed JWT.
+   */
+  defaultScopes?: string | string[];
```

Comment:
Huh - how would I differentiate this with the normal `scopes` property?

Reply:
`defaultScopes` are set so the auth library can fall back to using scopes if a self-signed JWT cannot be used. Situations where this happens: Using Metadata credentials or when specifying an API endpoint

Reply:
I don't see a user ever needing to set these, unless they were using `googleauth` with a custom API.

Reply:
Ah - the way this is set up though, a user *could* directly modify them. Having defaults as private members is fine, but exposing this publicly on the options is confusing to me at least.  But I concede ignorance here.

Reply:
Are you proposing `defaultScopes` be set in the constructor only? I agree if so!

Reply:
I'm open to moving  the `defaultScopes` to being only configurable within the constructor  👍 

The user could still define them here.

Reply:
I am confused :) Do we expect the user to ever set default scopes?  That name feels like something that's internal.

Reply:
We do not expect users to ever set default scopes

Reply:
In that case, this property shouldn't exist on `GoogleAuthOptions`, and it should be flagged `private` everywhere I think

Reply:
@JustinBeckwith this doesn't work because we need the client libraries themselves to set the option. What we're differentiating between is the default value set by the client, which itself instantiates the auth client, and scopes passed  from someone doing this:

```js
const {SecretManagerClient} = require('@google-cloud/secret-manager)
new SecretManagerClient({
  scopes: ['some-scopes']
})
```

☝️ this cannot be self signed.

**In the client itself:**

```js
const auth = new GoogleAuth({
 defaultScopes: ['secret-manager-scopes']
})
```

☝️ this should be self signed .

Reply:
Basically, we need to have  passed a value for the default scopes, from the generated client, through to [here](https://github.com/googleapis/gax-nodejs/blob/f7ebfb6fb3f7f1707debc7274f0af57382107ebc/src/grpc.ts#L96), which then needs to be used in instantiating auth client.